sqlglot/parser.py

```diff
@@ -6667,16 +6667,36 @@ def _parse_copy_parameters(self) -> t.List[exp.CopyParameter]:
 
         return options
 
+    def _parse_var_property(self) -> t.Optional[exp.Expression]:
+        prop = self._parse_property()
+
+        if not isinstance(prop, exp.Property):
+            return prop
+
+        # Replace the unquoted identifiers with vars
+        this = prop.this.find(exp.Identifier)
+        value = prop.args.get("value")
+        value = value.find(exp.Identifier) if value else None
+        if this and not this.quoted:
+            this.replace(exp.var(this.name))
+        if value and not value.quoted:
+            value.replace(exp.var(value.name))
+
+        return prop
+
     def _parse_credentials(self) -> t.Optional[exp.Credentials]:
         expr = self.expression(exp.Credentials)
 
         if self._match_text_seq("STORAGE_INTEGRATION", advance=False):
-            expr.set("storage", self._parse_conjunction())
+            expr.set("storage", self._parse_var_property())
         if self._match_text_seq("CREDENTIALS"):
-            # Snowflake supports CREDENTIALS = (...), while Redshift CREDENTIALS <string>
-            creds = (
-                self._parse_wrapped_options() if self._match(TokenType.EQ) else self._parse_field()
-            )
+            creds: t.Optional[exp.Expression] | t.List[t.Optional[exp.Expression]] = None
+            if self._match(TokenType.EQ):
+                # Snowflake case: CREDENTIALS = (...) . Note that empty parens are allowed too, thus the var fallback
+                creds = self._parse_wrapped_options() or [exp.Var(this=None)]
```

Comment:
I've defaulted to setting `creds = []` which will at least make it a non-None arg. Then the generator can use this as a proxy instead of generating an `exp.Var(None)`